src/llvm_backend_proc.cpp

```diff
@@ -2809,6 +2809,23 @@ gb_internal lbValue lb_build_builtin_proc(lbProcedure *p, Ast *expr, TypeAndValu
 			}
 			return res;
 		}
+	case BuiltinProc_read_cycle_counter_frequency:
+		{
+			lbValue res = {};
+			res.type = tv.type;
+
+			if (build_context.metrics.arch == TargetArch_arm64) {
+				LLVMTypeRef func_type = LLVMFunctionType(LLVMInt64TypeInContext(p->module->ctx), nullptr, 0, false);
+				bool has_side_effects = false;
+				LLVMValueRef the_asm = llvm_get_inline_asm(func_type, str_lit("mrs $0, cntfrq_el0"), str_lit("=r"), has_side_effects);
+				GB_ASSERT(the_asm != nullptr);
+				res.value = LLVMBuildCall2(p->builder, func_type, the_asm, nullptr, 0, "");
+			} else {
+				GB_PANIC("Unsupported architecture: %.*s", LIT(target_arch_names[build_context.metrics.arch]));
```

Comment:
I'm not sure this is the right move, but it's probably better than nothing?
Might be worth looking at our other arch-specific calls and adding unsupport arch there too, I don't think x86_cpuid / x86_xgetvb or friends check either, I used them for reference.

Reply:
A panic isn't the best idea though, we class compiler panics as compiler bugs (ie if a user hits a cpp panic/assert it's a compiler bug) and this seems like something you can hit quite easily. I think you'd want a check in the checker and emit a proper error there instead. `wasm_memory_grow` has such a check and error message for example.